lib/releaselib.sh

```diff
@@ -452,6 +453,40 @@ release::set_release_version () {
     logecho "RELEASE_VERSION_PRIME=$RELEASE_VERSION_PRIME"
   fi
 
+  # To ensure we're executing releases in the correct order, we append the keys
+  # of the RELEASE_VERSION associative array in the order that we explicitly
+  # prefer: official, rc, beta, alpha
+  #
+  # In anago, we will then iterate over the keys in the ORDERED_RELEASE_KEYS
+  # array and where necessary use these elements to access the underlying values
+  # of the RELEASE_VERSION associative array.
+  #
+  # Example:
+  #
+  # for release_type in "${ORDERED_RELEASE_KEYS[@]}"; do
+  #  echo "Doing the $release stuff (${RELEASE_VERSION[$release_type]})"
+  # done
+  #
+  #
+  # More on manipulating associative arrays:
+  # https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Arrays
+
+  if [[ ${RELEASE_VERSION[official]} ]]; then
+    ORDERED_RELEASE_KEYS+=("official")
+  fi
+
+  if [[ ${RELEASE_VERSION[rc]} ]]; then
+    ORDERED_RELEASE_KEYS+=("rc")
+  fi
+
+  if [[ ${RELEASE_VERSION[beta]} ]]; then
+    ORDERED_RELEASE_KEYS+=("beta")
+  fi
```

Comment:
Do we also need/want `beta0` & `beta1` here?

See: https://github.com/kubernetes/release/blob/72cf1a3c5f28f2a21538b23a2b91dc6e02e828d3/lib/releaselib.sh#L394-L407

Reply:
@hoegaarden -- would prefer to defer that to another PR post-release, as to not introduce any more uncertainty than necessary.

The PR as-is handles the `official` and `beta` code paths, which are the ones we care about for the patch releases.

I definitely did [notice this](https://twitter.com/stephenaugustus/status/1217305345699254272?s=20) though and plan to fix it.
Ideally we can find a way to rip it out, instead of hacking around it.

Reply:
While we never did that (skimming the code & doc this seems to refer to emergency releases?), I just wanted to make sure we do not it that in case we'd need/want to cut an emergency release. However, it seems we never needed to do this in the past (can't find any of those tags in k/k) and I am not sure if or under which circumstances we'd want to do that. So 👍 for ripping that out, especially 🆒 as people are at least aware of that weirdness.

/lgtm